processor/src/lib.rs

```diff
@@ -78,18 +92,26 @@ struct Process {
 }
 
 impl Process {
-    pub fn new(inputs: ProgramInputs) -> Self {
-        Self {
+    fn initialize(inputs: ProgramInputs, enable_trace: bool) -> Process {
+        Process {
             system: System::new(4),
             decoder: Decoder::new(),
-            stack: Stack::new(&inputs, 4),
+            stack: Stack::new(&inputs, 4, enable_trace),
             hasher: Hasher::new(),
             bitwise: Bitwise::new(),
             memory: Memory::new(),
             advice: AdviceProvider::new(inputs),
         }
     }
 
+    pub fn new(inputs: ProgramInputs) -> Self {
+        Self::initialize(inputs, false)
+    }
+
+    pub fn with_trace(inputs: ProgramInputs) -> Self {
```

Comment:
Technically, process always contains traces for various components (just not the overflow table). So, we probably should use a different name. Maybe `new_debug()`?